include/zephyr/drivers/mdio.h

```diff
@@ -54,6 +46,14 @@ __subsystem struct mdio_driver_api {
 	/** Write data to MDIO bus */
 	int (*write)(const struct device *dev, uint8_t prtad, uint8_t regad,
 		     uint16_t data);
+
+	/** Read data from MDIO bus using Clause 45 access */
+	int (*read_c45)(const struct device *dev, uint8_t prtad, uint8_t devad,
+			uint16_t regad, uint16_t *data);
+
+	/** Write data to MDIO bus using Clause 45 access */
+	int (*write_c45)(const struct device *dev, uint8_t prtad, uint8_t devad,
+			 uint16_t regad, uint16_t data);
```

Comment:
Should update the doxygen comments on the existing read/write functions to specify Clause 22 access

Reply:
In the context of this PR, the read/write functions are not only meant for Clause 22 access but also for proprietary protocols such as Micrel SMI, which is transmitted in the same 16-bit frame format of MIIM. Therefore I haven't updated the description to not limit it to Clause 22.